### PR TITLE
(HTCONDOR-2388)  Don't queue small transfers.

### PR DESCRIPTION
**Pull request by Todd-L-Miller**
Obviously still needs release notes, documentation, an entry in the param table, and some reason to believe it also works for uploads.

# HTCondor Pull Request Checklist for internal reviewers

- [ ] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [ ] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [ ] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [ ] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [ ] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [ ] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/condor_utils/file_transfer.cpp [hunk 1]
```diff
@@ -5724,9 +5724,19 @@ FileTransfer::DoObtainAndSendTransferGoAhead(DCTransferQueue &xfer_queue,bool do
...
-	if( !xfer_queue.RequestTransferQueueSlot(downloading,sandbox_size,full_fname,m_jobid.c_str(),queue_user.c_str(),timeout,error_desc) )
-	{
-		go_ahead = GO_AHEAD_FAILED;
+	// Don't bother to request a transfer queue slot for
+	// small-enough sandboxes.
+	long int min_required_to_transfer = param_integer(
+		"BYTES_REQUIRED_TO_QUEUE_FOR_TRANSFER", 100 * 1024 * 1024
+	);
+	if( sandbox_size <= min_required_to_transfer ) {
+		dprintf( D_ALWAYS, "Not entering transfer queue because sandbox is too small.\n" );
```
Comment (GregThain) [inline src/condor_utils/file_transfer.cpp:5733]: Can we dprintf the actual sandbox size and the limit here?

**Review by GregThain — CHANGES_REQUESTED** [review]

File: src/condor_utils/file_transfer.cpp (hunk 1 above)
Comment (Todd-L-Miller) [reply src/condor_utils/file_transfer.cpp]: Yes.

**Review by GregThain — APPROVED** [review]
LGTM.  I assume we are intentionally not documenting or version-history'ing this.

**Comment by Todd-L-Miller** [comment]
IIRC, we agreed that this was going in experimentally, and that if the experiment succeeds, we'll talk about how we want to share this feature with our users.  The JIRA tickets goes into a little more detail about what's missing.

**Pull request merged**